src/backend/core/api/viewsets.py

```diff
@@ -573,6 +573,12 @@ def retrieve(self, request, *args, **kwargs):
     @transaction.atomic
     def perform_create(self, serializer):
         """Set the current user as creator and owner of the newly created object."""
+        with connection.cursor() as cursor:
+            cursor.execute(
+                f'LOCK TABLE "{models.Document._meta.db_table}" '  # noqa: SLF001
+                "IN SHARE ROW EXCLUSIVE MODE;"
```

Comment:
You should add a comment explaining this mode.

Reply:
should be done when creating children as well? See [children method on the viewset](https://github.com/suitenumerique/docs/blob/main/src/backend/core/api/viewsets.py#L744).

Reply:
As discussed with lunika, I don't think the lock should be managed by the viewset, is should be at the model/manager level.
And as you mentioned, the same could happen to children/siblings, it should be covered also.

Reply:
We did try to add it in the model, see [here](https://github.com/suitenumerique/docs/compare/1140eef52f3ccb0ed94773e5cbfc5544d5e0fe6f?diff=unified&w=#diff-21170f63f26dc81c71cf947970745356a020118624a97a25988b67f4c40fdf3cR819-R833), but it brings instability in our tests with the factory, a fix brings another instability, and so on.

This naive way is much more stable.
Most of my flakiness is gone as well:
![image](https://github.com/user-attachments/assets/2f78e8ad-8764-4984-b960-45d0e29da94c)